src/main/java/com/zipcodewilmington/looplabs/StringDuplicateDeleter.java

```diff
@@ -5,4 +5,54 @@
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class StringDuplicateDeleter extends DuplicateDeleter<String> {
+
+    public StringDuplicateDeleter(String[] stringArray) {
+        super(stringArray);
+    }
+
+    @Override
+    public String[] removeDuplicatesExactly(int exactNumberOfDuplications) {
+        StringBuilder dupes = new StringBuilder();
+        int dupeCount = 0;
+        for (int i = 0; i < this.array.length; i++) {
+            for (int j = 0; j < this.array.length; j++) {
+                if (this.array[i].equals(this.array[j])) {
+                    dupeCount++;
+                }
+            }
+            if (!(dupeCount == exactNumberOfDuplications)) {
+                dupes.append(this.array[i]);
+                dupes.append(",");
+            }
+            dupeCount = 0;
+        }
+        String[] dupesRemoved = dupes.toString().split(",");
+        return dupesRemoved;
+    }
+
+
+    @Override
+    public String[] removeDuplicates(int maxNumberOfDuplications) {
+        StringBuilder dupes = new StringBuilder();
+        int dupeCount = 0;
+        for (int i = 0; i < this.array.length; i++) {
+            for (int j = 0; j < this.array.length; j++) {
+                if (this.array[i].equals(this.array[j])) {
+                    dupeCount++;
+                }
+            }
+            if (!(dupeCount >= maxNumberOfDuplications)) {
+                dupes.append(this.array[i]);
+                dupes.append(",");
+            }
+            dupeCount = 0;
+        }
+        String[] dupesRemoved = dupes.toString().split(",");
+        if (!dupesRemoved[0].equals("")) {
```

Comment:
while this isn't technically correct, your code will look better if you do your positive condition first. Generally the only time that you should use not in your if statement is if there is no else following it. 

This is fine though.